ethereumj-core/src/main/java/org/ethereum/sync/BlockDownloader.java

```diff
@@ -103,7 +103,15 @@ private void headerRetrieveLoop() {
                     }
                     int reqHeadersCounter = 0;
                     for (SyncQueueIfc.HeadersRequest headersRequest : hReq.split(MAX_IN_REQUEST)) {
-                        final Channel any = pool.getAnyIdle();
+
```

Comment:
`anyIdle()` randomness has some sense here: if for some reason we are assuming a peer to be best but it is invalid the sync may stuck since we will always be asking the same invalid peer.
M.b. it makes sense to add some randomness to `getBestIdle()`?